### DISCUSSION TITLE
Details on ObjectStoreBinder in v2.4.1

### DISCUSSION THREAD

**Discussion by jmichas** [shinyorg/shiny#875] [Q&A] [answered]
I just upgraded to 2.4.1. Prior to that I had problems with the `ObjectStoreBinder` stuff, it always generated an error at startup.
I had previously been using `RxApp.SuspensionHost` to create a custom suspend resume handler based around Akavache. This stores a global singleton with app info like username, uid, etc.

Now that I have 2.4.1 and the `ObjectStoreBinder` stuff seems to be working and picking up my global singleton and saving it to secure storage, do I need my `RxApp.SuspensionHost` anymore?

Also, when I change a property on a singleton being tracked by `ObjectStoreBinder` does it write that to storage immediately or is there a manual process to make sure it is persisted immediately?

Thanks!

**Comment by aritchie** [comment, marked as answer]
What was the problems you had with ObjectStoreBinder?  It has always been used because the internals of Shiny use it.  

I won't tell you what mechanism to use.  I would decide what works best for you and if changing your code is worth it.

ObjectStoreBinder will write immediately to the store upon detecting a change.

**Reply by jmichas** [reply]
I can't remember exactly what the issues were but I got a couple of exceptions at startup around not being able to bind objects. It was back on 2.1 or 2.2. I had put the mobile part aside to work on backend stuff, so it has been a while. I ended up pulling the source for the stores on iOS and tweaking them to remove the error, probably not the best move but it didn't appear to affect any functionality.

Anyway, I'd like to streamline around shiny as much as possible so if the rxapp suspension is doing the same thing or actually it seems to be a more cumbersome way of doing the same thing, then I would like to remove it. Also, the suspension host is saving to akavache secure store and shiny is saving the same object to another secure store, I will most likely have some race errors at some point if not already.

Good to know about the immediate write to the store, that helps my decision.

Thanks!
